lib/pubid_iso/parser.rb

```diff
@@ -26,29 +26,41 @@ class Parser < Parslet::Parser
         str("IS")).as(:stage)
     end
 
-    rule(:copublisher) do
-      (str("IEC") | str("IEEE") | str("IEC/IEEE") | str("CIW") | str("SAE") |
-        str("CIE") | str("ASME")).as(:copublisher)
-    end
-
     # TYPES = {
     #   "TS" => "technical-specification",
     #   "TR" => "technical-report",
     #   "PAS" => "publicly-available-specification",
     #   "Guide" => "guide",
     # }.freeze
     # DATA|GUIDE|ISP|IWA|PAS|R|TR|TS|TTA
+    #       # type          = "data" / "guide" / "isp" / "iwa" /
+    #       #   "pas" / "r" / "tr" / "ts" / "tta"
     rule(:type) do
-      (str("TS") | str("TR") | str("PAS") | str("Guide")).as(:type)
+      (str("DATA") | str("ISP") | str("IWA") | str("R") | str("TTA") |
+        str("TS") | str("TR") | str("PAS") | str("Guide")).as(:type)
     end
 
     rule(:year) do
       match('\d').repeat(4, 4)
     end
 
+    rule(:part) do
+      str("-") >> (match['[\dA-Z]'] | str("-")).repeat(1).as(:part)
+    end
+
+    rule(:originator) do
+      (str("ISO") | str("IWA")).as(:publisher) >> (str("/") >> copublisher).maybe
+    end
+
+    rule(:copublisher) do
```

Comment:
Remember IEC and IEEE are separate copublishers.

Reply:
> Remember IEC and IEEE are separate copublishers.

There is an identifier with IEC and IEEE together: "ISO/IEC/IEEE 26512"

Reply:
Yes, technically we have to treat that as 2 co-publishers. "IEC" and "IEEE" separately.

Reply:
This is now in #4